build/commands/lib/config.js

```diff
@@ -1065,6 +1065,7 @@ Object.defineProperty(Config.prototype, 'defaultOptions', {
     let env = Object.assign({}, process.env)
     env = this.addPathToEnv(env, path.join(this.depotToolsDir, 'python-bin'), true)
     env = this.addPathToEnv(env, path.join(this.depotToolsDir, 'python2-bin'), true)
+    env = this.addPathToEnv(env, path.join(this.srcDir, 'third_party', 'rust-toolchain', 'bin'), true)
```

Comment:
do you really need this? We should not add new things into `PATH` if possible.

Can't you use the full path in places where you need to call something from this dir?

Reply:
Yes, otherwise cargo can't find rustc or it will use local rustc which could be the wrong version. This ensures that we're always using the upstream rust toolchain

Reply:
we should remove `audit_deps` from gn files. It doesn't make sense to keep it there, because this target is only called from an npm command and it doesn't require any build dependency.

a follow up to cleanup this would be nice.

Reply:
> cargo can't find rustc or it will use local rustc which could be the wrong version

cargo is [supposed to respect](https://doc.rust-lang.org/cargo/reference/environment-variables.html) a `RUSTC` env variable. Maybe setting that in the build context would be less invasive than altering the path?